### DISCUSSION TITLE
How is ASTAP fed images from a OSC picture?

### DISCUSSION THREAD

**Discussion by mads0100** [mworion/MountWizzard4#237] [Q&A] [answered]
Hey Michael,

   I hope you are well.  There is a discussion on the indi forums that made me wonder if this is why I have to take exceptionally long (in my opinion) pictures compared to what I was used to with a mono camera in MW4.  Discussion here: https://indilib.org/forum/ekos/11388-announcing-stellarsolver-2-0.html

What channel is being pushed to ASTAP for platesolving from a OSC image in MW4? Is there a way to prioritize or give us an option when the camera is a OSC camera?  The consensus in the indi forum is that 'G' would make sense due to the nature of the CFA.  

Does this make sense or do you just pass the OSC image and I need to go talk with the ASTAP guy?

**Comment by mworion** [comment]
Hi,
the image transferees is the image provided by EKOS. In MW4 for processing it, I debayer the image and transform it to monochrome. From that on I use SEP algorithms only on monochrome. ASTAP gets the original picture for solving.
I checked the thread, there is no hint about longer exposure times needed. As I think ASTAP does not use SEP algorithms for source detection, so I guess it handles it different. From documentation there is no limit mentioned to OSC color channels.
Michel

**Comment by mads0100** [comment]
It is my experience that it takes longer exposures in order to correctly
platesolve in MW4 versus the internal solver in Ekos.  I've been grasping
at straws trying to figure it out.  I wondered if the monochrome conversion
might be a root cause for it.

**Comment by mworion** [comment]
Does ASTAP in MW4 not solve at all if you go for shorter exposures (which exposure time) or does solving does take just longer ?

> Am 17.05.2022 um 18:51 schrieb mads0100 ***@***.***>:
> 
> 
> It is my experience that it takes longer exposures in order to correctly
> platesolve in MW4 versus the internal solver in Ekos. I've been grasping
> at straws trying to figure it out. I wondered if the monochrome conversion
> might be a root cause for it.
>

**Comment by mads0100** [comment]
It doesn't solve at all.  I haven't found a way to log what ASTAP 'sees' so
I haven't been able to determine what the issue is.  I find if I use 20-30s
exposures, the pictures solve every time. Inside of Ekos, I can use 5s
exposures.  The only difference between the two is I can control gain
inside of Ekos (I use 100 for my particular camera, ASI2600).  I also
updated so I don't know if gain is working at this point inside MW4 too and
this issue is OBE.

I'd love to reduce the exposure time if possible.

**Comment by mworion** [comment]
OK, you should be able to control gain as well from MW4 (if the driver provides it over the remote interface). Just look to the imaging tab and check if gain is enabled.
If so, MW4 will set the gain for you.


> Am 17.05.2022 um 18:58 schrieb mads0100 ***@***.***>:
> 
> 
> It doesn't solve at all. I haven't found a way to log what ASTAP 'sees' so
> I haven't been able to determine what the issue is. I find if I use 20-30s
> exposures, the pictures solve every time. Inside of Ekos, I can use 5s
> exposures. The only difference between the two is I can control gain
> inside of Ekos (I use 100 for my particular camera, ASI2600). I also
> updated so I don't know if gain is working at this point inside MW4 too and
> this issue is OBE.
> 
> I'd love to reduce the exposure time if possible.
> 
>

**Comment by mads0100** [comment]
I will try the new version.  It's small potatoes, just trying to maximize
the shorter summer nights.  I think that might be the real culprit (0
versus 100 gain).

**Comment by mworion** [comment]
Please let me know how it worked out.

> Am 17.05.2022 um 19:03 schrieb mads0100 ***@***.***>:
> 
> 
> I will try the new version. It's small potatoes, just trying to maximize
> the shorter summer nights. I think that might be the real culprit (0
> versus 100 gain).
> 
> 
>

**Comment by mads0100** [comment]
Still can’t set gain or offset. Just shows a - and it’s not modifiable. What I see and what the driver shows. I’ll try binning but I don’t think it’s talking to indi either. 



Sent from my iPhone

On May 17, 2022, at 12:43, Michael ***@***.***> wrote:

﻿
Please let me know how it worked out.

> Am 17.05.2022 um 19:03 schrieb mads0100 ***@***.***>:
> 
> 
> I will try the new version. It's small potatoes, just trying to maximize
> the shorter summer nights. I think that might be the real culprit (0
> versus 100 gain).
> 
> 
>

**Comment by mads0100** [comment]
It’s also not passing that I set it to bin to Astap. 

Sent from my iPhone

On May 29, 2022, at 19:28, Chris Madson ***@***.***> wrote:

﻿Still can’t set gain or offset. Just shows a - and it’s not modifiable. What I see and what the driver shows. I’ll try binning but I don’t think it’s talking to indi either. 



Sent from my iPhone

On May 17, 2022, at 12:43, Michael ***@***.***> wrote:

﻿
Please let me know how it worked out.

> Am 17.05.2022 um 19:03 schrieb mads0100 ***@***.***>:
> 
> 
> I will try the new version. It's small potatoes, just trying to maximize
> the shorter summer nights. I think that might be the real culprit (0
> versus 100 gain).
> 
> 
>

**Comment by mads0100** [comment]
And strangely enough when I went to change it back to 1x1 in the imaging tab it set it to 2x2. 



Sent from my iPhone

On May 29, 2022, at 19:28, Chris Madson ***@***.***> wrote:

﻿Still can’t set gain or offset. Just shows a - and it’s not modifiable. What I see and what the driver shows. I’ll try binning but I don’t think it’s talking to indi either. 



Sent from my iPhone

On May 17, 2022, at 12:43, Michael ***@***.***> wrote:

﻿
Please let me know how it worked out.

> Am 17.05.2022 um 19:03 schrieb mads0100 ***@***.***>:
> 
> 
> I will try the new version. It's small potatoes, just trying to maximize
> the shorter summer nights. I think that might be the real culprit (0
> versus 100 gain).
> 
> 
>

**Comment by mworion** [comment, marked as answer]
Hi,
sorry to hear. I tested the topic in my Linux Setup (Ubuntu 20.04) with my QHY600 and a ASI178. The first supports Gain and Offset, the second not. Both could be set for bin. All properties are available in MW4 as they are in EKOS native. It would be very helpful, if you could post a log file from MW4 of you RASPI4 and please set the log level (under settings / log) to trace. This could definitely help finding out, what happens.
Michel

See the setup I experience:

![Bildschirmfoto 2022-05-30 um 16 09 00](https://user-images.githubusercontent.com/6632980/171011531-ffb0c04a-63a0-44a7-9c6f-1246d2781bc8.png)
![Bildschirmfoto 2022-05-30 um 16 09 10](https://user-images.githubusercontent.com/6632980/171011539-70b36705-2f2c-497c-97b6-e21be63993e2.png)
![Bildschirmfoto 2022-05-30 um 16 09 20](https://user-images.githubusercontent.com/6632980/171011543-40b635e8-3edb-40d9-91a1-912c069d38cc.png)
![Bildschirmfoto 2022-05-30 um 16 09 30](https://user-images.githubusercontent.com/6632980/171011545-08c5b7de-74cd-4315-9160-799a626f3d38.png)
![Bildschirmfoto 2022-05-30 um 16 13 54](https://user-images.githubusercontent.com/6632980/171011547-c23d8d6f-d4b5-459f-9113-0c75c29e5f34.png)
![Bildschirmfoto 2022-05-30 um 16 14 01](https://user-images.githubusercontent.com/6632980/171011551-0d57f2a8-1664-4d03-be4c-e556e9a0de18.png)
![Bildschirmfoto 2022-05-30 um 16 15 44](https://user-images.githubusercontent.com/6632980/171011555-ed60deb3-06c2-4a64-be43-5e367b581fea.png)
![Bildschirmfoto 2022-05-30 um 16 15 50](https://user-images.githubusercontent.com/6632980/171011558-e80c9d6f-7336-4e2c-8dcb-261de0dcb889.png)
